internal/executor/context/cluster_context_test.go

```diff
@@ -308,6 +313,33 @@ func TestKubernetesClusterContext_ProcessPodsToDelete_CallPatchOnClient_WhenPods
 	assert.True(t, exists)
 }
 
+func TestKubernetesClusterContext_ProcessPodsToDelete_ForceKill(t *testing.T) {
+	clusterContext, client := setupTest()
+	testClock := clock.NewFakeClock(time.Now())
+
+	// create a pod with a deletion event 5 minutes in the past. This should be ignored
+	pod := createSubmittedBatchPod(t, clusterContext)
+	pod.DeletionGracePeriodSeconds = pointer.Int64(10)
+	pod.DeletionTimestamp = &metav1.Time{Time: testClock.Now().Add(-300 * time.Second)}
+	client.Fake.ClearActions()
+
+	clusterContext.DeletePods([]*v1.Pod{pod})
+	clusterContext.ProcessPodsToDelete()
+	assert.Equal(t, len(client.Fake.Actions()), 0)
+
+	// create another pod with a deletion event 5 minutes and 10 seconds in the past. This should be force killed
+	pod = createSubmittedBatchPod(t, clusterContext)
+	pod.DeletionGracePeriodSeconds = pointer.Int64(10)
+	pod.DeletionTimestamp = &metav1.Time{Time: testClock.Now().Add(-310 * time.Second)}
+	client.Fake.ClearActions()
+
+	// submitPodsWithWait(t, clusterContext, pod)
+	client.Fake.ClearActions()
+	clusterContext.DeletePods([]*v1.Pod{pod})
+	clusterContext.ProcessPodsToDelete()
+	assert.Equal(t, len(client.Fake.Actions()), 2)
```

Comment:
Here I want to assert that we using a timout of 0 to kill the pod, but it doesn't look like the fake client tells me this.